test/Microsoft.ML.TestFramework/TestCommandBase.cs

```diff
@@ -2076,10 +2074,10 @@ public void CommandTrainingBinaryFieldAwareFactorizationMachineWithValidationAnd
             OutputPath outputPath = StdoutPath();
             string args = $"data={data} test={data} valid={data} in={model.Path} cont+" + " " + loaderArgs + " " + extraArgs;
             using (var newWriter = OpenWriter(outputPath.Path))
-            using (var env = new ConsoleEnvironment(42, outWriter: newWriter, errWriter: newWriter))
+            using (Env.RedirectChannelOutput(newWriter, newWriter))
             {
```

Comment:
just curious, why is this changing?

Reply:
We used to create a new env for every test. Now we should be using the common env (since scanning assemblies is now part of env setup, it'll be too expensive to do this scanning very often). But the env globally enumerates all channels, which means that the baseline output is going to be off (or even non-deterministic sometimes). Hence the change.

---
In reply to: [227571723](https://github.com/dotnet/machinelearning/pull/1350#discussion_r227571723) [](ancestors = 227571723)